Build fails on OpenBSD arm64

Hello, I'm unable to build geth on my Raspberry Pi 4 running OpenBSD -current. I'm not very knowledgeable on the Go build process. Please let me know what other information is necessary or if I should run any other commands.

#### System information

Raspberry Pi 4 (2GB RAM)
`$ uname -a`
`OpenBSD rpi4-openbsd.none 6.9 GENERIC.MP#1051 arm64`

#### Expected behaviour
geth builds successfully and exists in `./build/bin/`

#### Actual behaviour
geth fails on build, `./build/bin/` does not exist

#### Steps to reproduce the behaviour
`go install github.com/ethereum/go-ethereum/cmd/geth@latest` on a Raspberry Pi 4 running OpenBSD

#### ~~Backtrace~~ Logs
[goenv](https://github.com/ethereum/go-ethereum/files/6086989/goenv.txt)
[go install github.com/ethereum/go-ethereum/cmd/geth@latest](https://github.com/ethereum/go-ethereum/files/6087187/go-ethereum-build.txt) (the same errors occur when running `make geth` after cloning the git repo)

From your log:
```
# github.com/shirou/gopsutil/mem
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:21:17: undefined: CTLVm
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:21:24: undefined: VmUvmexp
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:26:14: undefined: sizeOfUvmexp
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:29:13: undefined: Uvmexp
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:43:17: undefined: CTLVm
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:43:24: undefined: VmUvmexp
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:48:14: undefined: sizeOfUvmexp
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:51:13: undefined: Uvmexp
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:72:16: undefined: CTLVfs
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:72:24: undefined: VfsGeneric
.go/pkg/mod/github.com/shirou/gopsutil@v2.20.5+incompatible/mem/mem_openbsd.go:72:24: too many errors
# github.com/ethereum/go-ethereum/cmd/utils
.go/pkg/mod/github.com/ethereum/go-ethereum@v1.10.0/cmd/utils/diskusage.go:34:19: stat.Bavail undefined (type unix.Statfs_t has no field or method Bavail)
.go/pkg/mod/github.com/ethereum/go-ethereum@v1.10.0/cmd/utils/diskusage.go:35:9: stat.Bavail undefined (type unix.Statfs_t has no field or method Bavail)
.go/pkg/mod/github.com/ethereum/go-ethereum@v1.10.0/cmd/utils/diskusage.go:41:37: stat.Bsize undefined (type unix.Statfs_t has no field or method Bsize)
```
Seems that there is a problem in gopsutil on OpenBSD and in "golang.org/x/sys/unix", 
We'll investigate it, thank you for reporting!

--- a/go.mod
+++ b/go.mod
@@ -45,13 +45,14 @@ require (
 	github.com/prometheus/tsdb v0.7.1
 	github.com/rjeczalik/notify v0.9.1
 	github.com/rs/cors v1.7.0
-	github.com/shirou/gopsutil v2.20.5+incompatible
+	github.com/shirou/gopsutil v3.21.3+incompatible
 	github.com/status-im/keycard-go v0.0.0-20190316090335-8537d3370df4
 	github.com/stretchr/testify v1.7.0
 	github.com/syndtr/goleveldb v1.0.1-0.20210305035536-64b5b1c73954
+	github.com/tklauser/go-sysconf v0.3.5 // indirect
 	github.com/tyler-smith/go-bip39 v1.0.1-0.20181017060643-dbb3b84ba2ef
 	golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
-	golang.org/x/sys v0.0.0-20210124154548-22da62e12c0c
+	golang.org/x/sys v0.0.0-20210316164454-77fc1eacc6aa
 	golang.org/x/text v0.3.4
 	golang.org/x/time v0.0.0-20201208040808-7e3f01d25324
 	gopkg.in/natefinch/npipe.v2 v2.0.0-20160621034901-c1b8fa8bdcce

--- a/go.mod
+++ b/go.mod
@@ -45,7 +45,7 @@ require (
 	github.com/prometheus/tsdb v0.7.1
 	github.com/rjeczalik/notify v0.9.1
 	github.com/rs/cors v1.7.0
-	github.com/shirou/gopsutil v3.21.3+incompatible
+	github.com/shirou/gopsutil v3.21.4-0.20210419000835-c7a38de76ee5+incompatible
 	github.com/status-im/keycard-go v0.0.0-20190316090335-8537d3370df4
 	github.com/stretchr/testify v1.7.0
 	github.com/syndtr/goleveldb v1.0.1-0.20210305035536-64b5b1c73954

